Change default upscaling to bilinear interpolation

tristanphease opened the pull request:
Fixes #142
Changes the default upscaling to instead of just taking the closest u/v chroma value average them out depending on their distance from the pixel, the same way that libwebp does it
Here's the bsk image from the issue decoded with this change
![image_output](https://github.com/user-attachments/assets/7292ff9a-68ab-42bc-aa35-68fb464d3b08)

Still need to work out whether we want the non fancy upscaling as a configuration option in the API (currently have it unused) and also the tests currently fail since the difference between the references and the output is now more than 10%

awxkee commented on src/yuv.rs:
```diff
@@ -0,0 +1,415 @@
+    }
+}
+
+fn get_fancy_chroma_value(main: u8, secondary1: u8, secondary2: u8, tertiary: u8) -> u8 {
```
It's better to mark all tiny methods as ``#[inline]` in hot points. They may already be inlined, but it's better to explicitly indicate that inlining is desired.

tristanphease commented:
Updated to incorporate some of the buffer changes from the other PR and also updated all the lossy reference images to be decoded with the default dwebp decoding settings and looks like it matches.

Remaining question I have is do we want to incorporate the non-fancy option in the public API for this crate so it's usable? We currently don't have any decoding parameters in decode_frame().

fintelia commented:
I'm in favor of switching the default to bilinear interpolation and adding a method `WebPDecoder` to switch back to the current approach.

Please do update tests/CREDITS.md with the precise command to produce the reference images.

fintelia reviewed (commented):
Looks good to me, just a few minor comments

tristanphease commented on src/lib.rs:
```diff
@@ -22,5 +22,6 @@ mod lossless;
 mod lossless_transform;
 mod transform;
 mod vp8_arithmetic_decoder;
+mod yuv;
```
It's in the vp8 module which is declared as `pub mod vp8`, might be more ergonomic to use if we export it in lib.rs rather than requiring users to use `image_webp::vp8::UpsamplingMethod`?

tristanphease replied:
Okay, changed to put it in decoder.rs so it can be used as `image_webp::UpsamplingMethod`. Also realised that this PR would be a breaking change since I changed the `Frame::set_rgb` definition which is actually public, is that a problem?

tristanphease commented on src/decoder.rs:
```diff
@@ -271,6 +272,29 @@ pub enum LoopCount {
 
+/// WebP decoder configuration options
+#[derive(Clone)]
+pub struct WebPDecodeOptions {
```
Thanks, changed
